core/src/main/java/org/kohsuke/stapler/jsr269/QueryParameterAnnotationProcessor.java

```diff
@@ -37,8 +40,21 @@ public boolean process(Set<? extends TypeElement> annotations, RoundEnvironment
                 }
             }
 
+            Map<String, String> output = new HashMap<>();
             for (ExecutableElement m : methods) {
-                write(m);
+                String paramNames = m.getParameters().stream()
+                        .map(VariableElement::getSimpleName)
+                        .collect(Collectors.joining(","));
+                String existing = output.get(m.getSimpleName().toString());
+                /*
+                 * Allow multiple methods to have the same name but different argument types as long as the arguments
+                 * have the same names. This allows deprecated StaplerRequest/StaplerResponse methods to coexist
+                 * alongside non-deprecated StaplerRequest2/StaplerResponse2 methods.
+                 */
+                if (existing == null || !existing.equals(paramNames)) {
+                    write(paramNames, m);
+                    output.put(m.getSimpleName().toString(), paramNames);
+                }
```

Comment:
Needed to fix a compilation error in core. Comes with an associated unit test that demonstrates the problem.